haystack_experimental/components/retrievers/multi_query_embedding_retriever.py

```diff
@@ -0,0 +1,152 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from concurrent.futures import ThreadPoolExecutor
+from typing import Any, List, Optional
+
+from haystack import Document, component, default_from_dict, default_to_dict
+from haystack.components.embedders.types.protocol import TextEmbedder
+from haystack.core.serialization import component_to_dict
+from haystack.utils.deserialization import deserialize_component_inplace
+
+from haystack_experimental.components.retrievers.types import EmbeddingRetriever
+
+
+@component
+class MultiQueryEmbeddingRetriever:
+    """
+    A component that retrieves documents using multiple queries in parallel with an embedding-based retriever.
+
+    This component takes a list of text queries, converts them to embeddings using a query embedder,
+    and then uses an embedding-based retriever to find relevant documents for each query in parallel.
+    The results are combined and sorted by relevance score.
+
+    ### Usage example
+
+    ```python
+    from haystack.components.embedders import SentenceTransformersQueryEmbedder
+    from haystack_experimental.components.retrievers import MultiQueryEmbeddingRetriever
+    from haystack_integrations.components.retrievers.opensearch import OpenSearchEmbeddingRetriever
+
+    query_embedder = SentenceTransformersQueryEmbedder(model="sentence-transformers/all-MiniLM-L6-v2")
+    retriever = OpenSearchEmbeddingRetriever(document_store=document_store)
```

Comment:
This example isn't runnable currently since `document_store` isn't defined. Could we switch it to an `InMemoryDocumentStore` and add some documents to the store to make the example runnable?